library/core/src/num/mod.rs

```diff
@@ -1414,11 +1414,11 @@ macro_rules! from_str_radix {
         impl $int_ty {
             /// Converts a string slice in a given base to an integer.
             ///
-            /// The string is expected to be an optional `+` sign
-            /// followed by digits.
-            /// Leading and trailing whitespace represent an error.
-            /// Digits are a subset of these characters, depending on `radix`:
+            /// The string is expected to be an optional `+` or `-` sign (the latter not being
+            /// allowed for unsigned integers) followed by only digits. Leading and trailing
```

Comment:
Probably best to just have different docs rather than showing this for unsigned numbers. You can move `sign_dependent_expr` (https://github.com/rust-lang/rust/blob/1afc5fd042f7583b9668dd62be98325487483d1c/library/core/src/num/nonzero.rs#L1695) to the parent `num/mod.rs`, and then adjust the `from_str_radix` matcher to take something like `unsigned u8, u16, u32, u64, u128; signed i8 i16 i32 i64 i128`.

Reply:
I was thinking the same but didn't know how to do conditional documentation in a macro. Many thanks for the pointers.

I've done it ~ish - my macro knowledge is not good enough to figure out how to do it with that exact syntax so I split it into two separate invocations instead. No problem with doing your exact suggestion if you could help me out with what pattern to use in the macro